packages/calling-stateful-client/src/CallSubscriber.ts

```diff
@@ -173,6 +165,16 @@ export class CallSubscriber {
 
   private stateChanged = (): void => {
```

Comment:
stateChanged() serves for its own purposes, let's not add unrelated code here

create a new private function named initCaptionSubscriber(), then add to stateChanged event:
```
this._call.on('stateChanged', this.initCaptionSubscriber);
```

And don't forget to unsubscribe the event after you creating the subscriber (also you will need to unsubscribe it in unsubscribe() function)